File: dressed-android/app/src/main/java/com/dressed/app/data/picker/PickerAIProvider.kt
package com.dressed.app.data.picker

/** BYOK cloud provider for wardrobe picker explanations. */
enum class PickerAIProvider(val storageKey: String) {
    ANTHROPIC("anthropic"),
    OPENAI("openai"),
    GROK("grok"),
    ;

    val displayName: String
        get() = when (this) {
            ANTHROPIC -> "Anthropic (Claude)"
            OPENAI -> "OpenAI"
            GROK -> "Grok / xAI"
        }

    val shortBannerLabel: String
        get() = when (this) {
            ANTHROPIC -> "Claude"
            OPENAI -> "OpenAI"
            GROK -> "Grok"
        }

    companion object {
        fun fromStorageKey(raw: String?): PickerAIProvider {
            if (raw == null) return ANTHROPIC
            return entries.find { it.storageKey == raw } ?: ANTHROPIC
        }
    }
}

File: dressed-ios/Dressed iOS/Dressed iOS/Sources/DressedKit/PickerAIProvider.swift
import Foundation

/// Cloud provider for wardrobe picker “AI reasoning” explanations (BYOK).
enum PickerAIProvider: String, CaseIterable, Identifiable {
    case anthropic
    case openai
    case grok

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .anthropic: return "Anthropic (Claude)"
        case .openai: return "OpenAI"
        case .grok: return "Grok / xAI"
        }
    }

    /// Short label for the picker status banner.
    var shortBannerLabel: String {
        switch self {
        case .anthropic: return "Claude"
        case .openai: return "OpenAI"
        case .grok: return "Grok"
        }
    }
}
